### DISCUSSION TITLE
Support taking pointers to fields of ref structs without pinning.

### DISCUSSION THREAD

**Discussion by teo-tsirpanis** [dotnet/csharplang#7066] [Language Ideas] [answered]
Consider the following snippet:

```csharp
public unsafe ref struct s {
    int _x;
    public int* X => &_x;
}
```

It currently fails with `error CS0212: You can only take the address of an unfixed expression inside of a fixed statement initializer`. But since `ref struct`s are constrained to reside on the stack, taking a pointer to one of their fields should be fine.

**Comment by CyrusNajmabadi** [comment]
Could you instead expose a ref to that int, and leave it to the caller of they want to treat it as a pointer?

**Comment by 333fred** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/issues/1792.
